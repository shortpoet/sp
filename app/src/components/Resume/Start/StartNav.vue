<template>
<nav
class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"
id="sideNav"
@click="collapse($event)"
>
<!-- $event apparently helps make sure jest sees the event -->
<a class="navbar-brand js-scroll-trigger" href="#about">
<span class="d-block d-lg-none">Resume</span>
<span id="pic" class="d-none d-lg-block">
<img class="img-fluid img-profile mx-auto mb-2" :src="pic" alt="profile pic" />
</span>
</a>
<button
class="navbar-toggler"
type="button"
data-toggle="collapse"
data-target="#navbarSupportedContent"
aria-controls="navbarSupportedContent"
aria-expanded="false"
aria-label="Toggle navigation"
>
<span class="navbar-toggler-icon"></span>
</button>
<div class="collapse navbar-collapse" id="navbarSupportedContent">
<ul class="navbar-nav">
<li class="nav-item">
<a class="nav-link js-scroll-trigger" href="#about">About</a>
</li>
<li class="nav-item">
<a class="nav-link js-scroll-trigger" href="#skills">Skills</a>
</li>
<li class="nav-item">
<a class="nav-link js-scroll-trigger" href="#awards">Natural Languages</a>
</li>
<li class="nav-item">
<a class="nav-link js-scroll-trigger" href="#experience">Experience</a>
</li>
<li class="nav-item">
<a class="nav-link js-scroll-trigger" href="#education">Education</a>
</li>
<li class="nav-item">
<a class="nav-link js-scroll-trigger" href="#interests">Interests</a>
</li>
</ul>
</div>
</nav>
</template>

<script>
export default {
name: "StartNav",
data() {
return {
pic: require("@/assets/github_profile_pic.jpg"),
isCollapse: false,
};
},
methods: {
collapse() {
console.log("collapse is called");
// Closes responsive menu when a scroll trigger link is clicked
// moving this here to be able to test
// added local boolean that creates coverage
// also makes it so the
this.$(".navbar-collapse").collapse("hide");
this.isCollapse = true;
},
/* istanbul ignore next */
revertPic() {
// @mouseenter="cyclePics" @mouseleave="revertPic"
// this.pic = require('@/assets/profile_pic.jpg')
// clearInterval()
},
/* istanbul ignore next */
cyclePics() {
// let i = 0
// let pics = [require('@/assets/profile_pic1.jpg'), require('@/assets/profile_pic2.jpg')]
// let counter = 0
// let cycle = () => {
//   if (i == 0) {
//     this.pic = pics[1]
//     i = 1
//   } else {
//     this.pic = pics[0]
//     i = 0
//   }
//   if (++counter > 5) {
//     clearInterval(cycle)
//   }
// }
// setInterval(cycle, 1500)
},
},
};
</script>
