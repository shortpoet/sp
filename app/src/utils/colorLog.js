const chalk = require('chalk')
import {inspect} from 'util'

// const _log  = () => Function.prototype.bind.call(console.log, console, 'LOG_ALIAS')

export const log = (color, message) => console.log(chalk[`${color}`](`${inspect(message)}`))

export const colorLog = (message, color, background) => {
  color = color || "black";
  if (background) {
    console.log(`%c` + `${message}`, `color:` + `${color};background:${background}`)
  } else {
    console.log(`%c` + `${message}`, `color:` + `${color}`)
  }
}

// module.exports =  {
//   log,
//   colorLog
// }
