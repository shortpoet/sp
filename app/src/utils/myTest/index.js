import createWrapper from './test.utils.createWrapper'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'
import propFinder from './test.utils.propFinder'
import propsFinder from './test.utils.propsFinder'
import propMocker from './test.utils.propMocker'
import propsMocker from './test.utils.propsMocker'
import textMatcher from './test.utils.textMatcher'
import textMatcherFactory from './test.utils.textMatcherFactory'

export default {
  createWrapper,
  factory,
  factoryShallow,
  propFinder,
  propsFinder,
  propMocker,
  propsMocker,
  textMatcher,
  textMatcherFactory
}
