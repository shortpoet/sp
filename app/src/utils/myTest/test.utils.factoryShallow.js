import { shallowMount } from '@vue/test-utils'

export const factoryShallow = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    return shallowMount(component, options)
  }

module.exports = factoryShallow