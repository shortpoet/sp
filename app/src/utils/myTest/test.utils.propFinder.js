import {hardResume} from '@/assets/resume.js'

export const propFinder = (prop) => {
  return {
    propsData: {
      [`${prop}`]: hardResume[`${prop}`]
    }
  }
}

module.exports = propFinder