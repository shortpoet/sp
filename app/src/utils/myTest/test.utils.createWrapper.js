import { createLocalVue } from '@vue/test-utils'
import factory from './test.utils.factory'
import factoryShallow from './test.utils.factoryShallow'

// import VueRouter from 'vue-router'
import Vuex from 'vuex'
import PortalVue from 'portal-vue'

import createStore from '@/store/createStore'
// import createMockRouterConfig from '@/router/createMockRouterConfig'
// import paths from '@/router/paths.js'

import { log } from '@/utils/colorLog'
export const createWrapper  = (
  component,
  /* istanbul ignore next */
  options = {},
  // but why don't i have to ignore the empty  or even unprovided resumeStoreOtions
  // might need to figure this out but setting params to ignore for now
  resumeStoreOptions = {},
  // routerOptions = {},
  isShallow = false,
  ) => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(PortalVue)
    const mockStoreResume = createStore.createMocks().createStoreResumeMocks(resumeStoreOptions)
    const storeConfig = createStore.createMocks({modules: {resume: mockStoreResume}})

    // Using router is untested makes tests fail
    // and just doesn't work for now
    // localVue.use(VueRouter)
    // const routerConfig = createMockRouterConfig.createMockRouterConfig(routerOptions)
    // const router = new VueRouter(routerConfig)

    // log('green', mockStoreResume)
    // log('yellow', storeConfig.modules.resume)
    const store = new Vuex.Store(storeConfig)
    // log('red', store)
    // log('blue', store._mutations['resume/SET_RESUME_RAW'][0]())
    // console.log(options)
    return isShallow ? 
    factoryShallow(component, {
      localVue,
      store,
      // router,
      ...options
    }) : 
    factory(component, {
      localVue,
      store,
      // router,
      ...options
    })
}

// export default createWrapper
// when this was set to export default a console log(createWrapper) that was in
// test.utils.textMatcher line 19 was being executed from every module that called createWrapper
module.exports = createWrapper