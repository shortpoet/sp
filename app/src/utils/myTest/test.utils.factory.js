import { mount } from '@vue/test-utils'
import { log } from '@/utils/colorLog'
export const factory = (
  component, 
  /* istanbul ignore next */
  options = {
    propsData: {},
    localVue: {},
    mocks: {},
    store: {},
    // context: {}, // only for functional components
    // this only breaks when testing
    // probably should comment out of real func (which this is (now) hehe)
    router: {},
    computed: {},
    stubs: {},
    slots: {},
    scopedSlots: {},
    attrs: {},
    attachToDocument: false,
    attachTo: null, //HTML Element or String
    listeners: {},
    parentComponent: {},
    provide: {}
  }) => {
    // console.log(options)
    // console.log(options.mocks)
    // console.log(options.store)
    // console.log(options.localVue)
    // log('blue', options.stubs)
    // log('green', options.propsData)
    // console.log(mount)
    return mount(component, options)
  }

module.exports = factory

// export default factory