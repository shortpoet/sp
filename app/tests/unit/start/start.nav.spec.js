import * as myTest from '@/utils/myTest'
const { createWrapper } = myTest.default
import StartNav from '@/components/Resume/Start/StartNav'
import { mount } from '@vue/test-utils'
import Vue from 'vue'
const chalk = require('chalk')
require('matchmedia-polyfill');
require('matchmedia-polyfill/matchMedia.addListener');

const {$} = require('jquery')
// require('bootstrap')
// const jMock = jest.mock('jquery', () => jest.fn())
// const $ = jest.fn()
// jMock.$ = $
// const collapse = jest.fn()
// jMock.$.collapse = collapse

const testWidths = [400, 900, 1000]

const mediaQueries = [
  "(max-width: 768px)",
  "(min-width: 768px)",
  "(max-width: 992px)",
  "(min-width: 992px)",
  // "screen and (max-width: 768px)",
  // "screen and (min-width: 768px)",
  // "screen and (max-width: 992px)",
  // "screen and (min-width: 992px)",
  // "screen and (min-height: 600px)",
  // "screen and (max-height: 400px)",
  // "screen and (orientation: landscape)",
  // "(min-resolution: 150dpi)",
  // "(max-resolution: 72dpi)"
]

const resizeWindow = (x, y) => {
  window.innerWidth = x;
  window.innerHeight = y;
  window.dispatchEvent(new Event('resize'));
}


describe('StartNav.vue', () => {
  const component = StartNav
  // console.log(Object.keys(component))
  // console.log(component.methods)
  let propsData
  let computed
  let stubs
  let mocks

  let methods = {}
  
  let resumeStoreOptions
  let ignoredElements

  let mountOptions
  let wrapper
  let emmitted


  beforeEach(async () => {

    // Object.defineProperty(window, 'matchMedia', {
    //   writable: true,
    //   value: jest.fn().mockImplementation(query => ({
    //     matches: false,
    //     media: query,
    //     onchange: null,
    //     addEventListener: jest.fn(),
    //     removeEventListener: jest.fn(),
    //     dispatchEvent: jest.fn(),
    //   })),
    // });    

    // https://github.com/testing-library/react-testing-library/issues/353#issuecomment-510074776
    window.resizeTo = function resizeTo(width, height) {
      Object.assign(this, {
        innerWidth: width,
        innerHeight: height,
        outerWidth: width,
        outerHeight: height,
      }).dispatchEvent(new this.Event('resize'))
    }

    jest.resetModules()
    jest.clearAllMocks()

    propsData = { 
      // name: 'test name' 
    }

    computed = {
      // getResumeLoaded: () => true
    }

    stubs = {
    }

    mocks = {
      $: jest.fn(() => {
        return {
          click: jest.fn(),
          scrollspy: jest.fn(),
          collapse: jest.fn()
        }
      }),
      // collapse: jest.fn()
    }

    resumeStoreOptions = { 
    }

    ignoredElements = [
      // 'portal'
    ]

    mountOptions = { propsData, mocks, computed, stubs, ignoredElements }

    // mountOptions.attachToDocument = true

    wrapper = createWrapper(component, mountOptions, resumeStoreOptions)

  })
  describe("start.nav.mediaQueries", () => {

  

    // https://stackoverflow.com/questions/57167525/how-to-set-media-queries-with-jest
    testWidths.forEach(tw => {

      mediaQueries.forEach(mq => {
  
  
        window.matchMedia = media => ({
          addListener: () => {},
          removeListener: () => {},
          // matches: media === '(min-width: 545px)',
          matches: media === mediaQueries[2],
        });
  
        // window.matchMedia = jest.fn().mockImplementation(query => {
        //   return {
        //     matches: query === '(min-width: 240px) and (max-width: 767px)' ? false:true ,
        //     media: '',
        //     onchange: null,
        //     addListener: jest.fn(),
        //     removeListener: jest.fn(),
        //   };
        // });
        it(`doesn't display pic at media query below 992px ${mq}`, async () => {      
          // window.matchMedia = jest.fn(() => {
          //   return {
          //     matches: true,
          //     addEventListener: jest.fn(),
          //     removeEventListener: jest.fn()
          //   }
          // })
          // console.log(window.matchMedia(mq))
          window.matchMedia(mq)
          // window.resizeTo(tw, 800)
          wrapper = createWrapper(component, mountOptions, resumeStoreOptions)
          resizeWindow(tw, 800)
          // window.matchMedia(mediaQueries[3])
          // global.window.innerWidth = tw
          // global.window.dispatchEvent(new Event('resize'))
          await Vue.nextTick()
          if (matchMedia(mq).matches) {
            // console.log(chalk.red(`it's a match for ${mq}`))
            // console.log(window.innerWidth)
          }
          expect(wrapper.html()).toMatchSnapshot()
          // if (tw < 993) {
          //   console.log(tw)
          //   console.log(wrapper.find('.d-lg-block'))
          //   expect(wrapper.find('.d-lg-block')).not.toBeTruthy()
          // }
        })
      })
    })
  })

  describe('startnav.collapsed', () => {
    it('calls collapse', async () => {
      // const spy = jest.spyOn(StartNav.methods, 'collapse')
      // const collapse = jest.fn()
      const _wrapper = mount(StartNav)
      // _wrapper.setMethods({collapse: collapse})
  
      // expect(wrapper).toMatchSnapshot()
      console.log(wrapper.vm.isCollapse)
      const button = wrapper.find('.nav-item')
      await button.trigger('click')
      // console.log(wrapper.vm.isCollapse)
      expect(wrapper.vm.isCollapse).toBe(true)
      // expect(collapse).toHaveBeenCalled()
  
    })
  
    it('closes responsive menu when a link is clicked', async () => {

      // // window.resizeTo(400, 600)
      // // window.resizeTo(1200, 800)
      
      // let toggler = wrapper.find('.navbar-toggler')
      
      // // expect(toggler.isVisible()).toBe(true)

      // // expect(wrapper.find('p')).toBe(false)

      // console.log(wrapper.vm.isCollapse)
      // // wrapper.find('li.nav-item').trigger('click')
      // const button = wrapper.find('button')
      // console.log(button)
      // button.trigger('click')
      // console.log(wrapper.vm.isCollapse)
      // // let el = $('a.js-scroll-trigger')
      // // console.log(el)
      // // console.log($.fn)
      // // console.log($.fn.collapse)
      
      // // wrapper.find('a.js-scroll-trigger[href*="#"]:not([href="#"])').trigger('click')
      // // const collapseSpy = jest.spyOn(el, 'collapse').mockImplementation(() => jest.fn())
      
      // // console.log($)
      // // console.log(Object.keys($))
      // // console.log($['fn'])
      // // console.log(component.methods)
      // // const collapseSpy = jest.spyOn(component.methods, 'collapse').mockImplementation(() => jest.fn())
      // const collapse = jest.fn()
      // wrapper.setMethods({collapse: collapse})
      
      // // el.click()
      // const p = wrapper.find('p')
      // console.log(p)
      // expect(p.text()).toMatch('Test')
      // // use snapshot to inspect click action
      // expect(wrapper.html()).toMatchSnapshot()

      // expect(collapse).toHaveBeenCalled()

    })



    it('matches snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

  })

})
