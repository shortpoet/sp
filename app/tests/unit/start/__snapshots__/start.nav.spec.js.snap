// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (max-width: 768px) 1`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (max-width: 768px) 2`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (max-width: 768px) 3`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (max-width: 992px) 1`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (max-width: 992px) 2`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (max-width: 992px) 3`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (min-width: 768px) 1`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (min-width: 768px) 2`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (min-width: 768px) 3`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (min-width: 992px) 1`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (min-width: 992px) 2`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue start.nav.mediaQueries doesn't display pic at media query below 992px (min-width: 992px) 3`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;

exports[`StartNav.vue startnav.collapsed matches snapshot 1`] = `
<nav id="sideNav" class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"><a href="#about" class="navbar-brand js-scroll-trigger"><span class="d-block d-lg-none">Resume</span> <span id="pic" class="d-none d-lg-block"><img src="" alt="profile pic" class="img-fluid img-profile mx-auto mb-2"></span></a> <button type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler"><span class="navbar-toggler-icon"></span></button>
  <div id="navbarSupportedContent" class="collapse navbar-collapse">
    <ul class="navbar-nav">
      <li class="nav-item"><a href="#about" class="nav-link js-scroll-trigger">About</a></li>
      <li class="nav-item"><a href="#skills" class="nav-link js-scroll-trigger">Skills</a></li>
      <li class="nav-item"><a href="#awards" class="nav-link js-scroll-trigger">Natural Languages</a></li>
      <li class="nav-item"><a href="#experience" class="nav-link js-scroll-trigger">Experience</a></li>
      <li class="nav-item"><a href="#education" class="nav-link js-scroll-trigger">Education</a></li>
      <li class="nav-item"><a href="#interests" class="nav-link js-scroll-trigger">Interests</a></li>
    </ul>
  </div>
</nav>
`;
